client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/HotrodCache.java

```diff
@@ -227,11 +218,7 @@ public void addReconnectListener(final ReconnectListener<HotrodCache<K, V>> list
 
         } else {
 
-            return Futures.create(() -> {
-                return cache
-                    .withFlags(Flag.FORCE_RETURN_VALUE)
-                    .removeWithVersionAsync(key, version);
-            });
+            return Futures.create(() -> cache.removeWithVersionAsync(key, version));
```

Comment:
When using that method in `removeCommandHandlingAdapterInstance()`, the return value is actually needed.

That means when removing the `FORCE_RETURN_VALUE` flag here, the `removeCommandHandlingAdapterInstance()` method also needs to be changed. I think we can do that here in 1.2.x - the information, whether removal succeeded or not is only logged here anyway.

Reply:
You are suggesting that the `FORCE_RETURN_VALUE` flag determines whether we always get back a boolean as the result of *removeWithVersionAsync*, right? The only documentation I have found on this is https://infinispan.org/docs/9.4.x/user_guide/user_guide.html#return_values_2 which doesn't mention the *removeWithVersionAsync* method so I assume that the flag has no impact on that method. Or have you conducted any particular tests that would suggest otherwise?

Reply:
I mixed it up with the `removeAsync` method here. So, you're right, I think the flag can be removed here without it affecting the return value.